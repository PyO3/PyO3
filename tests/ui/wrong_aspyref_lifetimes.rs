use pyo3::{types::PyDict, PyDetached, Python};

fn main() {
    let dict: PyDetached<PyDict> = Python::with_gil(|py| PyDict::new(py).into());

    // Should not be able to get access to Py contents outside of with_gil.
    let dict: &PyDict = Python::with_gil(|py| dict.as_ref(py));

    let _py: Python = dict.py(); // Obtain a Python<'p> without GIL.
}
