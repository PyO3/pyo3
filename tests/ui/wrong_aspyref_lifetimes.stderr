error: lifetime may not live long enough
 --> tests/ui/wrong_aspyref_lifetimes.rs:7:56
  |
7 |     let dict: &Bound<'_, PyDict> = Python::attach(|py| dict.bind(py));
  |                                                    --- ^^^^^^^^^^^^^ returning this value requires that `'1` must outlive `'2`
  |                                                    | |
  |                                                    | return type of closure is &'2 pyo3::Bound<'_, PyDict>
  |                                                    has type `Python<'1>`
